src/lib.rs

```diff
@@ -482,4 +484,31 @@ mod tests {
         assert_eq!(Some(vnode2), iter.next());
         assert_eq!(None, iter.next());
     }
+
+    struct TestIterStruct {
+        ring: HashRing<String>,
+    }
+
+    impl TestIterStruct {
+        pub fn print_each(&self) -> Vec<String> {
```

Comment:
Maybe `copy_each` would be a better name for this function since it's not printing the elements?